galsim/phase_psf.py

```diff
@@ -0,0 +1,845 @@
+                         of meters.  [Default: 0.2]
+    @param L0            Outer scale in meters.  The turbulence power spectrum will smoothly
+                         approach a constant at scales larger than L0.  Set to `None` or `np.inf`
+                         for a power spectrum without an outer scale.  [Default: 25.0]
+    @param vx            x-component wind velocity in meters/second.  [Default: 0.]
+    @param vy            y-component wind velocity in meters/second.  [Default: 0.]
+    @param alpha         Square root of fraction of phase that is "remembered" between time_steps
+                         (i.e., alpha**2 is the fraction remembered). The fraction sqrt(1-alpha**2)
+                         is then the amount of turbulence freshly generated in each step.  Setting
+                         alpha=1.0 results in a frozen-flow atmosphere.  Note that computing PSFs
+                         from frozen-flow atmospheres may be significantly faster than computing
+                         PSFs with non-frozen-flow atmospheres.  [Default: 1.0]
+    @param rng           Random number generator as a galsim.BaseDeviate().  If None, then use the
+                         clock time or system entropy to seed a new generator.  [Default: None]
+
+    Relevant SPIE paper:
+    "Remembrance of phases past: An autoregressive method for generating realistic atmospheres in
+    simulations"
+    Srikar Srinath, Univ. of California, Santa Cruz;
+    Lisa A. Poyneer, Lawrence Livermore National Lab.;
+    Alexander R. Rudy, UCSC; S. Mark Ammons, LLNL
+    Published in Proceedings Volume 9148: Adaptive Optics Systems IV
+    September 2014
+    """
+    def __init__(self, screen_size, screen_scale=None, altitude=0.0, time_step=0.03,
+                 r0_500=0.2, L0=25.0, vx=0.0, vy=0.0, alpha=1.0, rng=None):
+
+        if screen_scale is None:
+            screen_scale = 0.5 * r0_500
+        super(AtmosphericScreen, self).__init__(screen_size, screen_scale, altitude)
+
+        self.time_step = time_step
+        self.r0_500 = r0_500
+        self.L0 = L0
+        self.vx = vx
+        self.vy = vy
+        self.alpha = alpha
+
+        if rng is None:
+            rng = galsim.BaseDeviate()
+        self.orig_rng = rng
+
+        self._init_psi()
+
+        self.reset()
+
+        # Free some RAM for frozen-flow screen
+        if self.alpha == 1.0:
+            del self.psi, self.screen
+
+    def __str__(self):
+        return "galsim.AtmosphericScreen(altitude=%s)" % self.altitude
+
+    def __repr__(self):
+        outstr = ("galsim.AtmosphericScreen(%r, %r, altitude=%r, time_step=%r, " +
+                  "r0_500=%r, L0=%r, vx=%r, vy=%r, alpha=%r, rng=%r)")
+        return outstr % (self.screen_size, self.screen_scale, self.altitude, self.time_step,
+                         self.r0_500, self.L0, self.vx, self.vy, self.alpha, self.rng)
```

Comment:
You're missing the standard unit tests of picklability.  And I'm pretty sure this repr isn't going to pass.  (Not sure about the other serialization tests -- they might be ok.)  So far we've been making sure that `eval(repr(obj))` will correctly reconstruct any GalSim object, at least give the appropriate imports.  But this won't work here if the screen has been advanced at all.  

Now, maybe we don't need it to.  It's not unreasonable for the repr to just build the object as it started, but not necessarily how it exists later.  Our other classes mostly don't have internal states where this kind of thing matters.  But if you want to keep the repr test working, you can add private construction variables (with a leading underscore) to set origin, screen and tab2d appropriately for the current state.